### ISSUE TITLE
Configure errors in all 'ats2', 'sems-rhel6' and 'tlcc2' with BOOST_ROOT starting 2020-04-28

### ISSUE THREAD

**Issue by bartlettroscoe** [trilinos/Trilinos#7275]
CC: @gsjaardema 

<Checklist>
<???: Add label "client: ATDM">
<???: Add label "ATDM Sev: Blocker" (by default but could be other "ATDM Sev: XXX")>
<???: Add label "type: bug"?>
<???: Add label "impacting: tests"?>
<???: Add label for affected packages (e.g. "pkg: MueLu", "pkg: Tpetra", "pkg: Kokkos", etc.)>
<???: Add label "PA: ???Project Area???" (e.g. "PA: Linear Solvers", "PA: Data Services")>
<???: Add milestone "Initial cleanup of new ATDM ..." or "Keep promoted ATDM ...">
<???: Once GitHub Issue is created, add entries for tests to TrilinosATDMStatus/*.csv files>

## Next Action Status

<status-and-or-first-action>

## Description

As shown in [this query](https://testing-dev.sandia.gov/cdash/index.php?project=Trilinos&date=2020-04-28&filtercount=2&showfilters=1&filtercombine=and&field1=buildname&compare1=65&value1=Trilinos-atdm-&field2=configureerrors&compare2=43&value2=0) the there are configure errors in the builds:

* `Trilinos-atdm-ats2-cuda-10.1.243-gnu-7.3.1-spmpi-2019.06.24_static_dbg`
* `Trilinos-atdm-ats2-cuda-10.1.243-gnu-7.3.1-spmpi-2019.06.24_static_dbg_cuda-aware-mpi`
* `Trilinos-atdm-ats2-cuda-10.1.243-gnu-7.3.1-spmpi-2019.06.24_static_opt`
* `Trilinos-atdm-ats2-cuda-10.1.243-gnu-7.3.1-spmpi-2019.06.24_static_opt_cuda-aware-mpi`
* `Trilinos-atdm-ats2-gnu-7.3.1-spmpi-2019.06.24_serial_static_dbg`
* `Trilinos-atdm-ats2-gnu-7.3.1-spmpi-2019.06.24_serial_static_opt`
* `Trilinos-atdm-sems-rhel6-gnu-7.2.0-openmp-complex-shared-release-debugN`
* `Trilinos-atdm-sems-rhel6-gnu-7.2.0-openmp-debug	`
* `Trilinos-atdm-sems-rhel6-gnu-7.2.0-openmp-release	`
* `Trilinos-atdm-sems-rhel6-gnu-7.2.0-openmp-release-debug	`
* `Trilinos-atdm-sems-rhel6-intel-17.0.1-openmp-release	`
* `Trilinos-atdm-tlcc2-intel-debug-openmp	`
* `Trilinos-atdm-tlcc2-intel-opt-openmp`

started failing on testing day 2020-04-28.

The 'ats2' builds on 'vortex' (as shown [here](https://testing-dev.sandia.gov/cdash/build/5410174/configure)), the 'sems-rhel6' builds (as shown [here](https://testing-dev.sandia.gov/cdash/build/5410639/configure)), and the 'tlcc2' builds, (as shown [here](https://testing-dev.sandia.gov/cdash/build/5411199/configure)) as show configure errors startling like:

```
-- Reading in configuration options from cmake/std/atdm/ATDMDevEnv.cmake ...
CMake Error at /jenkins/slave/workspace/Trilinos-atdm-sems-rhel6-gnu-7.2.0-openmp-complex-shared-release-debug/Trilinos/cmake/tribits/core/utils/AssertDefined.cmake:79 (MESSAGE):
  Error, the variable ENV{BOOST_ROOT} is not defined!
Call Stack (most recent call first):
  cmake/std/atdm/ATDMDevEnvSettings.cmake:36 (ASSERT_DEFINED)
  cmake/std/atdm/ATDMDevEnv.cmake:1 (INCLUDE)
  /jenkins/slave/workspace/Trilinos-atdm-sems-rhel6-gnu-7.2.0-openmp-complex-shared-release-debug/Trilinos/cmake/tribits/core/package_arch/TribitsGlobalMacros.cmake:184 (INCLUDE)
  /jenkins/slave/workspace/Trilinos-atdm-sems-rhel6-gnu-7.2.0-openmp-complex-shared-release-debug/Trilinos/cmake/tribits/core/package_arch/TribitsProjectImpl.cmake:96 (TRIBITS_READ_IN_OPTIONS_FROM_FILE)
  /jenkins/slave/workspace/Trilinos-atdm-sems-rhel6-gnu-7.2.0-openmp-complex-shared-release-debug/Trilinos/cmake/tribits/core/package_arch/TribitsProject.cmake:93 (TRIBITS_PROJECT_IMPL)
  CMakeLists.txt:90 (TRIBITS_PROJECT)
```

The new commits that were pulled on testing day 2020-04-28 as shown, for example, [here](https://testing-dev.sandia.gov/cdash/viewNotes.php?buildid=5410174#!#note1).  From looking over that set of commits, it seems possible that the commit 09e765db0a from PR #7197 may have triggered this.

## Current Status on CDash

* [ATDM Trilinos builds for current testing day](https://testing-dev.sandia.gov/cdash/index.php?project=Trilinos&filtercount=1&showfilters=1&field1=buildname&compare1=65&value1=Trilinos-atdm-)

## Steps to Reproduce

One should be able to reproduce this failure on the machine <supported-atdm-machine> as described in:

* https://github.com/trilinos/Trilinos/blob/develop/cmake/std/atdm/README.md

For example, the exact commands to reproduce this issue on one of the 'tlcc2' machines should be:

```
$ cd <some_build_dir>/

$ source $TRILINOS_DIR/cmake/std/atdm/load-env.sh Trilinos-atdm-tlcc2-intel-debug-openmp

$ cmake \
 -GNinja \
 -DTrilinos_CONFIGURE_OPTIONS_FILE:STRING=cmake/std/atdm/ATDMDevEnv.cmake \
 -DTrilinos_ENABLE_TESTS=ON -DTrilinos_ENABLE_ALL_PACKAGES=ON \
 $TRILINOS_DIR
```

**Comment by bartlettroscoe** [comment]
@gsjaardema, it looks like the `BOOST_ROOT` env var got removed in several of the environment.sh files from looking at:

```
$ git log-short -w -S"export BOOST_ROOT" -1 --name-status 09e765d -- cmake/std/atdm/
09e765d "CONFIG: Update NetCDF and HDF5 TPL version (#7197)"
Author: Greg Sjaardema <gsjaardema@gmail.com>
Date:   Mon Apr 27 09:37:57 2020 -0600 (24 hours ago)

M       cmake/std/atdm/ats2/environment.sh
M       cmake/std/atdm/common/toss3/environment_tlcc2.sh
M       cmake/std/atdm/sems-rhel6/environment.sh
```

Did the configure of Trilinos pass for you on these systems when you tested them when testing PR #7197 ?

**Comment by bartlettroscoe** [comment]
@gsjaardema, this is right in front of me so I will go ahead and create a commit to take care of this.

**Comment by bartlettroscoe** [comment]
FYI: PR #7276 is waiting to be reviewed, approved, and merged ...

**Comment by bartlettroscoe** [comment]
FYI: Since the PR tester seems to be having problems and may not allow the merge of PR #7276 today, I have manually merged that branch to the 'atdm-nightly-manual-updates' branch so these changes should be pulled in to that 'atdm-nightly' branch later today and this should be in place for testing tomorrow.

Moving this to "In Review".

**Comment by bartlettroscoe** [comment]
For some reason, we are still getting BOOST_ROOT configure errors for the 'ats2' builds on 'vortex' today as shown [here](https://testing-dev.sandia.gov/cdash/index.php?project=Trilinos&date=2020-04-29&filtercount=1&showfilters=1&field1=buildname&compare1=65&value1=Trilinos-atdm-ats2) showing:

```
-- Reading in configuration options from cmake/std/atdm/ATDMDevEnv.cmake ...
CMake Error at /vscratch1/jenkins/vortex-slave/workspace/Trilinos-atdm-ats2-cuda-10.1.243-gnu-7.3.1-spmpi-2019.06.24_static_dbg/Trilinos/cmake/tribits/core/utils/AssertDefined.cmake:79 (MESSAGE):
  Error, the variable ENV{BOOST_ROOT} is not defined!
Call Stack (most recent call first):
  cmake/std/atdm/ATDMDevEnvSettings.cmake:36 (ASSERT_DEFINED)
  cmake/std/atdm/ATDMDevEnv.cmake:1 (INCLUDE)
  /vscratch1/jenkins/vortex-slave/workspace/Trilinos-atdm-ats2-cuda-10.1.243-gnu-7.3.1-spmpi-2019.06.24_static_dbg/Trilinos/cmake/tribits/core/package_arch/TribitsGlobalMacros.cmake:184 (INCLUDE)
  /vscratch1/jenkins/vortex-slave/workspace/Trilinos-atdm-ats2-cuda-10.1.243-gnu-7.3.1-spmpi-2019.06.24_static_dbg/Trilinos/cmake/tribits/core/package_arch/TribitsProjectImpl.cmake:96 (TRIBITS_READ_IN_OPTIONS_FROM_FILE)
  /vscratch1/jenkins/vortex-slave/workspace/Trilinos-atdm-ats2-cuda-10.1.243-gnu-7.3.1-spmpi-2019.06.24_static_dbg/Trilinos/cmake/tribits/core/package_arch/TribitsProject.cmake:93 (TRIBITS_PROJECT_IMPL)
  CMakeLists.txt:90 (TRIBITS_PROJECT)
```

I will see how this can be happening and get this fixed.

**Comment by bartlettroscoe** [comment]
The reason we still have these errors is because the fixes did not get merged to the 'atdm-nightly' branch last night.  For details, see [ATDV-344](https://sems-atlassian-srn.sandia.gov/browse/ATDV-344?focusedCommentId=57423&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-57423).  We will need to wait one more day to see that the fixes in PR #7276 actually fix the problems.

**Comment by bartlettroscoe** [comment]
I believe with the merging of recent commits and PRs, all of these configure errors are now gone. If you look at the ATDM Trilinos builds today shown [here](https://testing-dev.sandia.gov/cdash/index.php?project=Trilinos&date=2020-05-05&filtercount=1&showfilters=1&field1=buildname&compare1=65&value1=Trilinos-atdm-) you see just configure failures in some Intel builds as shown [here](https://testing-dev.sandia.gov/cdash/index.php?project=Trilinos&date=2020-05-05&filtercount=2&showfilters=1&filtercombine=and&field1=buildname&compare1=65&value1=Trilinos-atdm-&field2=configureerrors&compare2=43&value2=0).  And if you look at thse you can see they are all due to unrelated Intel license sever failures (see [ATDV-346](https://sems-atlassian-srn.sandia.gov/browse/ATDV-346)).

Therefore, I am going to close this as complete.
